modules.cmake

```diff
@@ -72,5 +72,20 @@ function(add_module_library name)
         DEPENDS ${pcm})
     endforeach ()
   endif ()
+
+  # Common: add sources
   target_sources(${name} PRIVATE ${sources})
+
+  # MSVC
+  if (MSVC)
+    foreach (src ${sources})
+        # compile file as interface
+        set_source_files_properties(${src} PROPERTIES COMPILE_FLAGS /interface)
```

Comment:
Why not use target_compile_opitons? Currently we treat all sources the same.

Reply:
to resolve a situation similar to this one

```cmake
add_module_library(cool_lib module_file.cc)
target_sources(cool_lib non_module_file.cc)
```

The `target_compile_options` will change the behavior for all files, and this may not be what the caller wants.